torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1171,12 +1171,21 @@ def clip_grad_norm_(
         Returns:
             Total norm of the parameters (viewed as a single vector).
 
-        .. note:: This is analogous to ``torch.nn.utils.clip_grad_norm_`` but
-            handles the partitioning and multiple devices per rank under the
-            hood. The default torch util is not applicable here, because each
-            rank only has a partial view of all the grads in the model, so
-            calling it for FSDP models would lead to different scaling being
-            applied per subset of model parameters.
+        .. note:: If every FSDP instance uses ``NO_SHARD``, meaning that no
+            gradients are sharded across ranks, then you may directly use
+            :func:`torch.nn.utils.clip_grad_norm_`.
```

Comment:
Can we warn explicitly about this?

Reply:
I have it so that if all instances use `NO_SHARD`, then this method returns `torch.nn.utils.clip_grad_norm_()`, so it is equivalent.